include/swift/AST/DiagnosticsSIL.def

```diff
@@ -295,6 +295,27 @@ ERROR(wrong_non_negative_assumption,none,
 ERROR(shifting_all_significant_bits,none,
       "shift amount is greater than or equal to type size in bits", ())
 
+// Float-point overflows and underflows
+WARNING(float_trunc_overflow, none,
+  "floating-point literal '%0' overflows %1-bit floating-point type",
+  (StringRef, unsigned int))
+
+WARNING(float_trunc_underflow, none,
+  "floating-point literal '%0' is subnormal and is not precisely "
+  "representable in a %1-bit floating-point type",
+  (StringRef, unsigned int))
+
+WARNING(float_nonrep, none, "input literal is outside the range of largest "
+  "representable floating-point literal", ())
+
+ERROR(float_to_int_overflow, none,
+  "floating-point literal '%0' overflows %1-bit %2 integer type",
```

Comment:
For %2 here (signed vs. unsigned) we usually use a %select. See arithmetic_operation_overflow_generic_type for an example rather than hard-coding string constants at the call site. This makes diagnostics easier to localize.